src/Core/Entity/Image/InputImage.php

```diff
@@ -70,6 +70,41 @@ public function __construct(OptionsBag $optionsBag, string $sourceImageUrl)
         }
     }
 
+    /**
+     * Properly encode URL
+     *
+     * @param string $url
+     * @return string
+     */
+    private function encodeUrl($url)
+    {
+        // URL scheme was not always set as the input scheme
+        if (preg_match("!https?:/[a-zA-Z]!", $url)) {
+            $url = preg_replace("!http(s?):/!", 'http$1://', $url);
+        }
+        $parsedUrl = parse_url($url);
+
+        if (empty($parsedUrl['scheme']) || empty($parsedUrl['host'])) {
+            return $url; // invalid URL
+        }
+
+        $user = isset($parsedUrl['user']) ? rawurlencode($parsedUrl['user']) : '';
+        $pass = isset($parsedUrl['pass']) ? ':' . rawurlencode($parsedUrl['pass'])  : '';
+        $auth = $user || $pass ? "$user$pass@" : '';
+
+        $host = $parsedUrl['host'];
+        $port = isset($parsedUrl['port']) ? ':' . $parsedUrl['port'] : '';
+
+        $path = '';
+        if (isset($parsedUrl['path'])) {
+            $path = implode('/', array_map('rawurlencode', explode('/', $parsedUrl['path'])));
+        }
+        $query = isset($parsedUrl['query']) ? '?' . $parsedUrl['query'] : '';
+        $fragment = isset($parsedUrl['fragment']) ? '#' . $parsedUrl['fragment'] : '';
+
+        return  $parsedUrl['scheme'] . '://' . $auth . $host . $port . $path . $query . $fragment;;
```

Comment:
@TimWeyand can you please fix this line? there's a double `;;` at the end.
Thanks!